meeting-notes/2016-12-01.md

```diff
@@ -0,0 +1,308 @@
+# VM WG Summit
+
+* Date: 2016-12-01 (Collaborators’ Summit - Austin)
+* Issue: <https://github.com/nodejs/summit/issues/19>
+* Notes: <https://docs.google.com/document/d/1D7-RBkuJiakADIWXOBVNyb6FNSXdoE3XIEXZ1axq97Q>
+
+
+## Attendees
+
+* Arunesh Chandra (@aruneshchandra)
+* Josh Gavant (@joshgav)
+* Matteo Collina (@mcollina)
+* CJ Ihrig (@cjihrig)
+* Jeremiah Senkpiel (@Fishrock123)
+* Rod Vagg (@rvagg)
+* Taylor Woll (@boingoing)
+* Hitesh Kanwathirtha (@digitalinfinity)
+* James Snell (@jasnell)
+* Stephen Belanger (@qard)
+* Michael Dawson (@mhdawson)
+* Mark Marron (@mrkmarron)
+* Bert Belder (@piscisaureus)
+* Myles Borins (@thealphanerd)
+* JeongHoon Byun (@outsideris)
+* Seth Thompson (@s3ththompson)
+* Charlie Robbins (@indexzero)
+* Dan Shaw (@dshaw)
+
+
+## Agenda
+
+* Node.js Native API (NAPI)
+* FFI
+* VM compliance spec
+* Next steps
+
+
+## Node.js Native API (NAPI)
+
+* [Repo](https://github.com/nodejs/abi-stable-node)
+* [EP](https://github.com/nodejs/node-eps/pull/20)
+* [Roadmap](https://github.com/nodejs/abi-stable-node/issues/18)
+
+
+General roadmap:
+
+1. Stable NAPI for use with native modules. NAN is currently used by many native
+modules; NAPI is addressing weaknesses of NAN abstractions, specifically
+leaking V8 APIs.
+2. Adoption by native modules in the ecosystem.
+3. Adoption by builtin modules in core. These directly use V8 APIs, will
+eventually need to use NAPI.
+
+@rvagg: TJ worked on a stable abstraction for a native ABI, but was too
+difficult because of changes in V8. How confident are we that this ABI will
+actually be stable for a long term, e.g. 12 months?
+
+@mhdawson: Goal has been to prove that current ABI works with 0.10, 0.12, 4.0,
+6.0, and even ChakraCore. Some adjustments were needed, but it wasn’t that
+difficult to hide changes between versions. Proving that this approach works
+
+@rvagg: What about significant future changes in V8? Do we have their buy-in?
+
+@jasnell: Whose responsibility is it to ensure the VM APIs continue to work with
+NAPI? Node’s or the VM owner’s?
+
+@mhdawson: Depends on needs of Node and each VM.
+
+We provide an `Environment*` pointer in each NAPI API, which gives deeper access
+if necessary.
+
+@aruneshchandra: How do we ensure buy-in from VM owners?
+
+@rvagg: If NAPI is accepted downstream VM owners will have to participate.
+
+@s3ththompson: V8 team has reviewed NAPI work and doesn’t have major concerns,
```

Comment:
Thanks for writing up these notes! This summary doesn't quite capture what I was trying to convey.  Perhaps: "the V8 team haven't yet uncovered any major issues, but we also haven't done a full review of the proposal."

I think it's under next steps below that the V8 team needs to do a review of the API.

cc @jeisinger